ttg/ttg/madness/ttg.h

```diff
@@ -323,6 +325,52 @@ namespace ttg_madness {
     cacheT cache;
 
    protected:
+    template <typename terminalT, std::size_t i, typename Key>
+    void invoke_pull_terminal(terminalT &in, const Key &key, TTArgs *args) {
+      if (in.is_pull_terminal) {
+        auto owner = in.container.owner(key);
+        if (owner != world.rank()) {
+          get_terminal_data<i, Key>(owner, key);
+        } else {
+          auto value = (in.container).get(key);
+          if (args->nargs[i] == 0) {
+            ::ttg::print_error(world.rank(), ":", get_name(), " : ", key,
+                               ": error argument is already finalized : ", i);
+            throw std::runtime_error("Op::set_arg called for a finalized stream");
+          }
+
+          if (typeid(value) != typeid(std::nullptr_t) && i < std::tuple_size_v<input_values_tuple_type>) {
+            this->get<i, std::decay_t<decltype(value)> &>(args->input_values) =
+              std::forward<decltype(value)>(value);
+            args->nargs[i] = 0;
+            args->counter--;
+          }
+        }
+      }
+    }
+
+    template <std::size_t i, typename Key>
+    void get_terminal_data(const int owner,
+                           const Key &key) {
+      if (owner != world.rank()) {
```

Comment:
Is there any case where `get_terminal_data` is called with `owner == world.rank()`? I only see the case here https://github.com/TESSEorg/ttg/pull/191/files#diff-90cb53d7fdb7d2978ed2afc294cb0ba485aaa2b246183c8840bedfbf5d14a0e5R333

Reply:
Yes, in case of local data, owner will be same as world.rank(). Wavefront example has data distributed across processes, so some data is local and some is remote.